pylint/plugins/hass_imports.py

```diff
@@ -338,14 +338,18 @@ def visit_importfrom(self, node: nodes.ImportFrom) -> None:
         ):
             self.add_message("hass-relative-import", node=node)
             return
-        if (
-            self.current_package.startswith("homeassistant.components")
-            and node.modname == "homeassistant.components"
-        ):
-            for name in node.names:
-                if name[0] == self.current_package.split(".")[2]:
-                    self.add_message("hass-relative-import", node=node)
-            return
+        if self.current_package.startswith("homeassistant.components."):
+            current_component = self.current_package.split(".")[2]
+            if node.modname == "homeassistant.components":
+                for name in node.names:
+                    if name[0] == current_component:
+                        self.add_message("hass-relative-import", node=node)
+                return
+            if node.modname.startswith(
+                f"homeassistant.components.{current_component}."
```

Comment:
Using a trailing dot to prevent issues when a components name start with the name of another component (eg. `homekit` and `homekit_controller`)

Reply:
Can you add a small test case for it? Just so we don't accidentally introduce a regression later.